src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -99,9 +99,11 @@ public static object GetUninitializedObjectWithFormatterServices(Type type)
         private static List<IExternalSerializer> externalSerializers;
         private static ConcurrentDictionary<Type, IExternalSerializer> typeToExternalSerializerDictionary;
         private static Dictionary<string, Type> types;
-        private static Dictionary<RuntimeTypeHandle, DeepCopier> copiers;
-        private static Dictionary<RuntimeTypeHandle, Serializer> serializers;
-        private static Dictionary<RuntimeTypeHandle, Deserializer> deserializers;
+        volatile private static Dictionary<RuntimeTypeHandle, DeepCopier> copiers;
+        volatile private static Dictionary<RuntimeTypeHandle, Serializer> serializers;
+        volatile private static Dictionary<RuntimeTypeHandle, Deserializer> deserializers;
```

Comment:
On the volatile keyword, I am worried here that you won't see the reference change. The readers have no lock and no memory barrier. So if a thread has cached the reference, will the lock on another thread be enough to ensure it gets the new reference or could it in theory be cached to the old reference for ever? Also as the serializer and deserializer can only ever be as a pair, is it worth just making it a dictionary<type,Tuple<Serilizer,Deserizlizer>> ?